src/data-import.ts

```diff
@@ -0,0 +1,331 @@
+  REPLACE = 'replace', // Replace entire inventory
+  MERGE = 'merge', // Merge quantities (add to existing)
+  SKIP_EXISTING = 'skip_existing', // Only import new colors
+}
+
+export interface ImportPreview {
+  newColors: Array<{
+    code: string;
+    quantity: number;
+  }>;
+  updatedColors: Array<{
+    code: string;
+    currentQuantity: number;
+    newQuantity: number;
+    finalQuantity: number;
+  }>;
+  unchangedColors: Array<{
+    code: string;
+    quantity: number;
+  }>;
+  totalChanges: number;
+  metadata: {
+    importedAt: string;
+    totalColors: number;
+    importSource: string;
+  };
+}
+
+export interface ImportResult {
+  success: boolean;
+  message: string;
+  preview?: ImportPreview;
+  error?: string;
+}
+
+/**
+ * Check if File System Access API is supported
+ */
+export function isFileSystemAccessSupported(): boolean {
+  return typeof window !== 'undefined' && 'showOpenFilePicker' in window;
+}
+
+/**
+ * Open file picker using File System Access API or fallback
+ */
+export async function openFilePicker(): Promise<File | null> {
+  if (isFileSystemAccessSupported()) {
+    try {
+      const [fileHandle] = await (window as any).showOpenFilePicker({
+        types: [
+          {
+            description: 'Montana Inventory Files',
+            accept: {
+              'application/json': ['.json', '.txt'],
+              'text/plain': ['.json', '.txt'],
+            },
+          },
+        ],
+        excludeAcceptAllOption: true,
+        multiple: false,
+      });
+      
+      return await fileHandle.getFile();
+    } catch (error) {
+      if ((error as Error).name === 'AbortError') {
+        // User cancelled - not an error
+        return null;
+      }
+      throw error;
+    }
+  } else {
+    // Fallback to traditional file input
+    return new Promise((resolve) => {
+      const input = document.createElement('input');
+      input.type = 'file';
+      input.accept = '.json,.txt,application/json,text/plain';
+      
+      input.onchange = (event) => {
+        const file = (event.target as HTMLInputElement).files?.[0];
+        resolve(file || null);
+      };
+      
+      input.oncancel = () => resolve(null);
+      
+      input.click();
+    });
+  }
+}
+
+/**
+ * Parse and validate import file content
+ */
+export async function parseImportFile(file: File): Promise<ImportData> {
+  if (!file) {
+    throw new Error('Nessun file selezionato');
+  }
+
+  // Check file size (max 10MB for safety)
+  if (file.size > 10 * 1024 * 1024) {
+    throw new Error('File troppo grande. Dimensione massima: 10MB');
+  }
+
+  let content: string;
+  try {
+    content = await file.text();
+  } catch (error) {
+    throw new Error('Errore nella lettura del file');
+  }
+
+  if (!content.trim()) {
+    throw new Error('Il file è vuoto');
+  }
+
+  let data: any;
+  try {
+    data = JSON.parse(content);
+  } catch (error) {
+    throw new Error('Il file non contiene JSON valido');
+  }
```

Comment:
I would bubble this error up as well, so that it's the consumer's responsibility to handle eventual errors when invoking `parseImportFile`